### ISSUE TITLE
URL recognition fails in tmux/screen splits

### ISSUE THREAD

**Issue by shazow** [jwilm/alacritty#1705]
Regarding the recent #1617 which added click-to-open URLs.

When a URL is wrapped inside of a tmux/screen split, the recongition pattern truncates it so that:

![image](https://user-images.githubusercontent.com/6292/47513641-adcff000-d84c-11e8-80c5-d4b9d20151a6.png)

Will open `https://github.com/jwilm/alacritty/blob/maste|`

How to reproduce:
1. Run `tmux` in alacritty.
2. `ctrl+b %` a couple times to make some vertical splits.
3. Paste echo a URL that is longer than the available width so that it wraps.
4. Try to click-open it.

Actual use case:
- I typically run IRC and other chat apps in a remote tmux with vertical splits. I'd like to be able to click-open links people share easily.

Solution ideas:
- Maybe an optional `url.wrap_chars`?
- To reduce the likelihood of a wrap symbol being in a valid URL, could check if it exists above/below the current wrap symbol.

**Comment by chrisduerr** [comment]
I do not think it is smart to support automatically trying to wrap. In a lot of scenarios this will likely just cause even more issues, since Alacritty might actually include parts into the URL which are completely unrelated.

This would also make it impossible to open valid URLs which end right at the `|` character, unless the next line starts with an empty cell.

However, what can be done, is that URL parsing will just stop when reaching the `|` character. However since `|` is a valid URL character, I don't think that's a good solution either.

Do you know if any other terminal emulator handles this differently? Otherwise I'd mark this as #wontfix, because there is no 'proper' solution to resolve this. A proper solution would probably be to support https://github.com/jwilm/alacritty/issues/922.

**Comment by shazow** [comment]
What about the solution I proposed?

> To reduce the likelihood of a wrap symbol being in a valid URL, could check if it exists above/below the current wrap symbol.

Also to emphasize, this would be _optional_.

Should keep an eye on how #992 interacts with splits, too.

**Comment by chrisduerr** [comment]
I'm not quite sure what you mean with your proposition. Could you maybe reword it?

**Comment by shazow** [comment]
In pseudocode-ish: if char at current_pos is in wrap_chars, then check terminal grid in current_pos.y-1 and current_pos.y+1 for wrap_chars also.

In other words, we assume that wrap characters are vertically contiguous. If we detect a wrap character, then we can check if there is another same wrap character directly above it or directly below it (in the sense of a vertical grid.

Positive case:

```
somelongurl| <- detected wrap character
moreurl....|...
```

```
somelongurl|
moreurl....| <- check for another wrap character below, yup looks good!
```

Negative case:

```
somelongurl|thathasapip
random other content
```

```
somelongurl| <- detected wrap character
random other <- not a wrap character, nevermind.
```

**Comment by chrisduerr** [comment]
This would not solve all the issues I've mentioned, but it would introduce a lot of additional complexity to the URL parsing.

I don't think this is a good area for Alacritty to try and innovate, unless there are existing solutions for this problem that work well (and I'm not aware of that), Alacritty should strive to support proper escape sequences to handle URLs which are broken across lines rather than Hacks which aren't going to work properly half of the time and just introduce a lot of false-positives.

**Comment by shazow** [comment]
I'm not sure which issues it wouldn't solve, and it's just a few extra lines of code with the grid datastructure that already exists, and this doesn't have to be defined by default so it wouldn't affect anyone negatively. Anyways, sounds like your mind is made up, feel free to wontfix it.

**Comment by jwilm** [comment]
Not going to implement this. The correct solution is using the URL escape sequences: #922.

**Comment by chrisduerr** [comment]
The problem is that it's still not obvious if a link ended at the end of the line, or if the link should continue at the next line. Also it's not clear if the separator is really a separator just by comparing to the previous/next line.

It's also dangerous to use configuration options as an excuse to include unused features, since a complex configuration might actually negatively affect the user (and Alacritty's configuration file already has a ton of stuff).

Please note that a proper solution to this problem would be great, you're certainly not the only one who wants to find a solution for this problem. However it's a complicated matter and a lot of times the separating character might not even be clearly defined.

I think we should strive to make escape sequences like the one discussed in #922 more popular, rather than making things worse by trying to work around the issue in a hacky way and introducing more problems than solutions.

**Comment by untitaker** [comment]
fwiw iterm2 does this with very few problems. I am not entirely sure *what* it does, but it works in a way that also works with vim splits etc
